pinot-core/src/main/java/com/linkedin/pinot/core/startree/StarTreeIndexNodeV2.java

```diff
@@ -0,0 +1,326 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.linkedin.pinot.core.startree;
+
+import java.util.Iterator;
+import java.util.Map;
+import xerial.larray.buffer.LBufferAPI;
+
+
+public class StarTreeIndexNodeV2 implements StarTreeIndexNodeInterf {
+  private static final int INVALID_INDEX = -1;
+
+  // Hard-code the size instead of computing at runtime, unit test ensures it matches actual size.
+  private static final int size = Integer.BYTES * 7;
```

Comment:
Can you make a constant for this magic number 7. I guess it is the number of elements the node of type INT. It will be good to make it a const so that it is more readable.